servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java

```diff
@@ -864,4 +837,82 @@ private static boolean validateHeaderValue(final byte value) {
+                "! / # / $ / % / & / ' / * / + / - / . / ^ / _ / ` / | / ~ / DIGIT / ALPHA";
+
+        private final CharSequence token;
+        private final String what;
+        private int idx;
+
+        TokenValidatingProcessor(final CharSequence token, final String what) {
+            this.token = token;
+            this.what = what;
+        }
+
+        @Override
+        public boolean process(final byte value) {
+            // HEADER
+            // header-field   = field-name ":" OWS field-value OWS
+            // field-name     = token
+            // token          = 1*tchar
+            //
+            //  Delimiters are chosen
+            //   from the set of US-ASCII visual characters not allowed in a token
+            //   (DQUOTE and "(),/:;<=>?@[\]{}")
+            //
+            // COOKIE
+            // cookie-pair       = cookie-name "=" cookie-value
+            // cookie-name       = token
+            // token          = 1*<any CHAR except CTLs or separators>
+            // CTL = <any US-ASCII control character
+            //       (octets 0 - 31) and DEL (127)>
+            // separators     = "(" | ")" | "<" | ">" | "@"
+            //                      | "," | ";" | ":" | "\" | <">
+            //                      | "/" | "[" | "]" | "?" | "="
+            //                      | "{" | "}" | SP | HT
+            //
+            // field-name's token is equivalent to cookie-name's token, we can reuse the tchar mask for both:
+            if (!isTchar(value)) {
+                throw new IllegalCharacterException(message(value));
+            }
+            ++idx;
+            return true;
+        }
+
+        private String message(final byte value) {
+            final int codePoint = value & 0xff;
+            final int idx = this.idx;
+            final StringBuilder sb = new StringBuilder(40 + stringSize(idx) + what.length() + token.length() +
+                    EXPECTED.length())
+                    .append('\'')   // 1
+                    .append((char) codePoint)   // 1
+                    .append("' (0x")    // 5
+                    .append(toHexString(0x100 | codePoint), 1, 3)   // 2 digit hex number
+                    .append(") at index=")  // 11
+                    .append(idx)
+                    .append(" in ") // 4
+                    .append(what)
+                    .append(' ')    // 1
+                    .append('\'')   // 1
+                    .append(token)
+                    .append("', expected [")    // 13
+                    .append(EXPECTED)
+                    .append(']');   // 1
+            return sb.toString();
+        }
+
+        private static int stringSize(final int value) {
+            assert value >= 0;
+            int size = 0;
+            int tmp = 1;
+            while (tmp <= value && tmp <= 1_000_000_000) {
+                ++size;
+                tmp = (tmp << 3) + (tmp << 1);
+            }
+            return size;
+        }
```

Comment:
At a glance it's not obvious what this is doing. What if you just gave it 6 bytes and let it resize in the exceedingly uncommon situation that we are wrong?

Alternatively, if you really want to count digits, would something like this be more readable?
```
int size = 1;
int test = 10;
while (test <= value && test  <= 1_000_000_000) {
  size++;
  test *= 10;
}
```

Reply:
Updated mehod name, added javadoc, renamed local variables, added a comment to clarify what shift does